Sources/Navigation/Extensions/WKNavigationDelegatePrivate.swift

```diff
@@ -0,0 +1,40 @@
+//  WKNavigationDelegatePrivate.swift
+//
+//  Copyright © 2023 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import WebKit
+
+@objc public protocol WKNavigationDelegatePrivate: WKNavigationDelegate {
```

Comment:
I’m trying to hunt down private API warnings found in the macOS PR. While I think those are in general not critical (we’re not _calling_ private WebKit symbols) I tried making this protocol internal. It doesn’t cause private API warnings to disappear, but the code still compiles.

Do you think it would make sense to make this protocol (and its implementation in DistributedNavigationDelegate) internal just to not expose more API from BSK?

Reply:
The intention of the protocol was making it public to keep the private API declarations in one place to avoid typos and for easier overriding from responder classes if needed

Reply:
Still it can be made private if you think it's better